backblazeb2/src/main/scala/akka/stream/alpakka/backblazeb2/scaladsl/B2API.scala

```diff
@@ -0,0 +1,235 @@
+
+    requestAndParse[AuthorizeAccountResponse](request)
+  }
+
+  /**
+   * https://www.backblaze.com/b2/docs/b2_get_upload_url.html
+   */
+  def getUploadUrl(
+      authorizeAccountResponse: AuthorizeAccountResponse,
+      bucketId: BucketId
+  ): B2Response[GetUploadUrlResponse] = {
+    val apiUrl = authorizeAccountResponse.apiUrl
+    val accountAuthorization = authorizeAccountResponse.authorizationToken
+    val uri = Uri(s"$apiUrl/$version/b2_get_upload_url").withQuery(Query("bucketId" -> bucketId.value))
+    val headers = RawHeader("Authorization", accountAuthorization.value)
+    val request = HttpRequest(
+      uri = uri,
+      method = HttpMethods.GET
+    ).withHeaders(headers)
+
+    requestAndParse[GetUploadUrlResponse](request)
+  }
+
+  /**
+   * https://www.backblaze.com/b2/docs/b2_upload_file.html
+   */
+  def uploadFile(
+      uploadCredentials: GetUploadUrlResponse,
+      fileName: FileName,
+      data: ByteString,
+      contentType: ContentType = DefaultContentType
+  ): B2Response[UploadFileResponse] = {
+    val uri = Uri(uploadCredentials.uploadUrl.value)
+    val headers =
+    RawHeader("Authorization", uploadCredentials.authorizationToken.value) ::
+    RawHeader("X-Bz-File-Name", B2Encoder.encode(fileName.value)) ::
+    RawHeader("X-Bz-Content-Sha1", B2Encoder.sha1String(data)) ::
+    Nil
+
+    val entity = HttpEntity(data).withContentType(contentType)
+
+    val request = HttpRequest(
+      uri = uri,
+      method = HttpMethods.POST
+    ).withHeadersAndEntity(headers, entity)
+
+    requestAndParse[UploadFileResponse](request)
+  }
+
+  private def requestAndParse[T: FromResponseUnmarshaller](request: HttpRequest): B2Response[T] = {
+    val requestId = UUID.randomUUID()
+    val start = System.nanoTime()
+    logger.trace(s"request $requestId: Invoking ${request.uri}")
+
+    Http()
+      .singleRequest(request)
+      .map { x =>
+        val end = System.nanoTime()
+        val delta = (end - start) / 1000000
+        logger.debug(s"request $requestId: Invoking ${request.uri} done in $delta ms with status ${x.status}")
+        x
+      }
+      .flatMap { response =>
+        parseResponse[T](response).recover {
+          case t: Throwable => // this adds useful debug info to the error
+            throw new RuntimeException(s"Failed to decode $response for $request", t)
+        }
+      }
+  }
+
+  private def authorizationHeaders(authorization: Option[AccountAuthorizationToken]) =
+    authorization.map(authorizationHeader).toSeq
+
+  private def authorizationHeader(authorization: AccountAuthorizationToken) =
+    RawHeader("Authorization", authorization.value)
+
+  /**
+   * https://www.backblaze.com/b2/docs/b2_download_file_by_name.html
+   */
+  def downloadFileByName(
+      fileName: FileName,
+      bucketName: BucketName,
+      apiUrl: ApiUrl,
+      accountAuthorization: Option[AccountAuthorizationToken]
+  ): B2Response[DownloadFileResponse] = {
```

Comment:
Here also the `DownloadFileResponse` has a field `data` which should be a `Source[ByteString, ...]` so downloading a file as a stream is possible.